segmentation/utils/spatial_analysis.py

```diff
@@ -195,3 +180,77 @@ def calculate_cluster_spatial_enrichment(all_data, dist_mats, fovs=None,
+
+
+def create_neighborhood_matrix(all_data, dist_matrices, fov_list=None, distlim=50, fov_col="SampleID",
+                               flowsom_col="FlowSOM_ID", cell_label_col="cellLabelInImage", cell_type_col="cell_type"):
+    """Calculates the number of neighbor phenotypes for each cell. The function counts each cell as one of its own
+    neighbors in this implementation.
+
+        Args:
+            all_data: data for the all fovs in the form of a pandas DF, including the columns of SampleID (fovs),
+                cellLabelInImage (the cell label), and FlowSOM_ID (the cell phenotype id).
+            dist_matrices: A dictionary that contains a cells x cells matrix with the euclidian
+                distance between centers of corresponding cells for every fov
+            fov_list: patient labels to include in analysis. If argument is none, default is all labels used.
+            distlim: cell proximity threshold. Default is 50.
+            fov_col: column with the cell fovs (Default is SampleID)
+            flowsom_col: column with the cell phenotype IDs (Default is FlowSOM_ID)
+            cell_label_col: column with the cell labels (Default is cellLabelInImage)
+            cell_type_col: column with the cell types (Default is cell_type)
+        Returns:
+            cell_neighbor_counts: matrix with phenotype counts per cell
+            cell_neighbor_freqs: matrix with phenotype frequencies of
+                counts per phenotype/total phenotypes for each cell
+            cell_count: current cell in analysis"""
+
+    # Setup input and parameters
+    if fov_list is None:
+        fov_list = sorted(list(set(all_data[fov_col])))
+
+    # Error Checking
+    if not np.isin(fov_list, all_data[fov_col]).all():
+        raise ValueError("Points were not found in Expression Matrix")
+
+    # Get the phenotypes
+    pheno_titles = all_data[cell_type_col].drop_duplicates()
+
+    # Subset just the sampleID, cellLabelInImage, and FlowSOMID
+    all_data = all_data[[fov_col, cell_label_col, flowsom_col, cell_type_col]]
+    # Extract the columns with the cell phenotype codes
+    pheno_codes = all_data[flowsom_col].drop_duplicates()
+    # Get the total number of phenotypes
+    pheno_num = len(pheno_codes)
+
+    # initiate empty matrices for cell neighborhood data
+    cell_neighbor_counts = pd.DataFrame(np.zeros((all_data.shape[0], pheno_num + 2)))
+
+    # Replace the first and second columns of cell_neighbor_counts with the fovs and cell labels respectively
+    cell_neighbor_counts[[0, 1]] = all_data[[fov_col, cell_label_col]]
+
+    # Rename the columns to match cell phenotypes
+    cols = [fov_col, cell_label_col] + list(pheno_titles)
+    cell_neighbor_counts.columns = cols
+
+    cell_neighbor_freqs = cell_neighbor_counts.copy(deep=True)
+
+    for i in range(len(fov_list)):
+        # Subsetting expression matrix to only include patients with correct label
+        patient_idx = all_data.iloc[:, 0] == fov_list[i]
```

Comment:
Correct me if I'm wrong, but I believe patient and FOV are actually referring to the same thing here? If so, let's change all references to patients in variable names to be FOVs so that we have consistency and the code is easier to follow